AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateEditor.java

```diff
@@ -636,33 +631,56 @@ private boolean deletionWouldOrphanNote(Collection col, TemporaryModel tempModel
             return false;
         }
 
-        ActivityResultLauncher<Intent> activityResultLauncher = registerForActivityResult(
-                new ActivityResultContracts.StartActivityForResult(),
-                new ActivityResultCallback<ActivityResult>() {
-                    @Override
-                    public void onActivityResult(ActivityResult result) {
-                        if (result.getResultCode() == REQUEST_CARD_BROWSER_APPEARANCE) {
-                            onCardBrowserAppearanceResult(result.getResultCode(), result.getData());
-                            return;
-                        }
+        public class onCardBrowserAppearanceContract extends ActivityResultContract<Intent, Intent>{
+            @NonNull
+            @Override
+            public Intent createIntent(@NonNull Context context, Intent input) {
+                return input;
+            }
 
-                        if (result.getResultCode() == REQUEST_PREVIEWER) {
-                            TemporaryModel.clearTempModelFiles();
-                            // Make sure the fragments reinitialize, otherwise there is staleness on return
-                            ((TemplatePagerAdapter)mTemplateEditor.mViewPager.getAdapter()).ordinalShift();
-                            mTemplateEditor.mViewPager.getAdapter().notifyDataSetChanged();
-                        }
-                    }
+
+            @Override
+            public Intent parseResult(int resultCode, @Nullable Intent intent) {
+                if(resultCode != Activity.RESULT_OK || intent == null) {
```

Comment:
If intent is null, then `return intent` is equivalent to `return null;`

```suggestion
                if (resultCode != Activity.RESULT_OK) {
```